src/ORM/Rule/LinkConstraint.php

```diff
@@ -0,0 +1,193 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ORM\Rule;
+
+use Cake\Datasource\EntityInterface;
+use Cake\ORM\Association;
+use Cake\ORM\Table;
+
+/**
+ * Checks whether links to a given association exist / do not exist.
+ */
+class LinkConstraint
+{
+    /**
+     * Link status that requires a link to be present.
+     *
+     * @var string
+     */
+    public const LINK_STATUS_LINKED = 'linked';
```

Comment:
`LINK_` prefix might not be necessary in scope of LinkConstraint class.

Reply:
Yeah, I'm not really sure what the conventions for that are... gotta take a look at the const usages in the core.